python/mxnet/executor.py

```diff
@@ -323,6 +325,20 @@ def output_dict(self):
                 self._symbol.list_outputs(), self.outputs)
         return self._output_dict
 
+    @property
+    def optimized_symbol(self):
```

Comment:
When would an user want to access optimized_symbols? Is this added only for testing purpose? Does it make sense to keep it private?

Reply:
@eric-haibin-lin For now this is added for testing only, but it's actually quite useful, and not just for the TensorRT integration, but potentially for other acceleration libraries in the future. Imagine wanting to visualize a model to figure out what the acceleration library did. Looking at the timeline may be helpful, but only if one knows which kernels belong to the accelerator, and which to MXNet or direct cuBLAS/cuDNN calls. Seeing how the graph got rewritten can help determine which layers could not be handled by the acceleration library, to determine what asks to make of libraries going forward. For example, TensorRT doesn't handle non-maximum suppression (NMS) directly, it can do that via a plugin. This initial integration doesn't handle TensorRT plugins. Knowing what didn't get subsumed into the accelerator graph node and comparing against a profile can determine whether for instance layers that are not being handled by the acceleration library subgraph are dominating the profile - a visual inspection makes this really quick. This could be useful for both the Graphviz integration, as well as potentially for TensorBoard. We tested this with GraphViz already, and it's working fine. Check out the attached visualization of LeNet-5. The graph on the left is the original MXNet one, and the one on the right is with all layers except for the input tensor having been replaced by a single TensorRT node. For a more complex network such as SSD, it would be visible that the NMS layer is a separate NNVM node, outside of the TensorRT subgraph. 

![trt](https://user-images.githubusercontent.com/476135/43219647-c285314c-8ffc-11e8-9903-bbb64b88ff3a.png)